nft/ethereum_payment_obligation.go

```diff
@@ -160,18 +161,41 @@ func (s *ethereumPaymentObligation) OwnerOf(registry common.Address, tokenID []b
 }
 
 // sendMintTransaction sends the actual transaction to mint the NFT
-func (s *ethereumPaymentObligation) sendMintTransaction(cid identity.CentID, contract ethereumPaymentObligationContract, opts *bind.TransactOpts, requestData *MintRequest) (*uuid.UUID, *types.Transaction, error) {
-	tx, ethTx, err := s.ethClient.SubmitTransaction(cid, contract.Mint, opts, requestData.To, requestData.TokenID, requestData.TokenURI, requestData.AnchorID,
+func (s *ethereumPaymentObligation) sendMintTransaction(
+	cid identity.CentID,
+	contract ethereumPaymentObligationContract,
+	opts *bind.TransactOpts,
+	requestData *MintRequest,
+	registry common.Address,
+	documentID []byte) (uuid.UUID, *types.Transaction, error) {
+	tx, err := s.ethClient.SubmitTransactionWithRetries(contract.Mint, opts, requestData.To, requestData.TokenID, requestData.TokenURI, requestData.AnchorID,
 		requestData.MerkleRoot, requestData.Values, requestData.Salts, requestData.Proofs)
+	if err != nil {
+		return uuid.Nil, nil, err
+	}
+
+	var txID uuid.UUID
+	txID, err = ethereum.QueueTransactionStatusTask(cid, tx.Hash(), s.txService, s.queue, func(status transactions.Status) error {
+		if status == transactions.Failed {
+			return nil
+		}
+
+		_, err := documents.InitNFTCreatedTask(
```

Comment:
I thought we are not going to do callbacks ;) 
(I totally agree with this change as I suggested :)